libs/pika/async_mpi/include/pika/async_mpi/dispatch_mpi.hpp

```diff
@@ -60,7 +60,9 @@ namespace pika::mpi::experimental::detail {
 #if defined(PIKA_HAVE_STDEXEC)
 
         using completion_signatures = pika::execution::experimental::completion_signatures<
-            pika::execution::experimental::set_value_t(MPI_Request)>;
+            pika::execution::experimental::set_value_t(MPI_Request),
+            pika::execution::experimental::set_error_t(std::exception_ptr),
```

Comment:
This is not quite accurate, as it could be forwarding another error type from previous senders. I can update this separately from your PR if you'd like, but wanted to point it out here.

Reply:
is this the correct way to do it ?
```
        using completion_signatures =
            pika::execution::experimental::transform_completion_signatures_of<std::decay_t<Sender>,
                pika::execution::experimental::empty_env,
                pika::execution::experimental::completion_signatures<
                    pika::execution::experimental::set_error_t(std::exception_ptr)>,
                pika::execution::experimental::completion_signatures<
                    pika::execution::experimental::set_value_t(MPI_Request),
                    pika::execution::experimental::set_stopped_t()>>;
```

Reply:
Close, but not quite, I think.

This is right, you want to add `set_error_t(exception_ptr)`:
```
                pika::execution::experimental::completion_signatures<
                    pika::execution::experimental::set_error_t(std::exception_ptr)>,
```
You could/should move the `set_stopped_t()` signature into this `completion_signatures`.

The `set_value_t` is wrong, I think. 

You'd want something like this:
```
        template <typename...>
        using no_value_completion = completion_signatures<>;
        using completion_signatures =
            pika::execution::experimental::transform_completion_signatures_of<std::decay_t<Sender>,
                pika::execution::experimental::empty_env,
                pika::execution::experimental::completion_signatures<
                    pika::execution::experimental::set_value_t(MPI_Request),
                    pika::execution::experimental::set_error_t(std::exception_ptr),
                    pika::execution::experimental::set_stopped_t() // this also gets added by the default SetDone template parameter
                >,
                no_value_completion>;
```
You're essentially saying I definitely want these `set_value/error/done` completion signatures, and I don't want any of the value completions from the previous sender. Implicitly you're also saying I want additionally all the error completions from the previous sender (it's the default `SetError`: https://github.com/NVIDIA/stdexec/blob/cddcd5a17dbec59d8502c0c6108553a20e74d2da/include/stdexec/__detail/__transform_completion_signatures.hpp#L319-L320).

See https://github.com/NVIDIA/stdexec/blob/cddcd5a17dbec59d8502c0c6108553a20e74d2da/include/stdexec/__detail/__transform_completion_signatures.hpp#L308-L369 for details.